Tests/EventBroadcasterTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿using NUnit.Framework;
+using UnityEngine;
+
+namespace DUCK.Utils.Tests
+{
+	[TestFixture]
+	public class EventBroadcasterTests
+	{
+		private interface IEventHandler
+		{
+			void HandleTheEvent();
+		}
+
+		private class TestBehaviour : MonoBehaviour, IEventHandler
```

Comment:
This wont work in 2018.2 and above. The Test assembly is classed as an editor folder, once you run the test it will inform you that you cant use that MonoBehaviour as it is in a Editor folder. See HierarchyBehaviour for the work around.

Reply:
I just ran the tests in 2018.2. Everything passed without any warning or error

Reply:
Yeah so did mine at first but then I released it and it all fell apart. I also managed to trick Unity into thinking it the assembly was okay and the tests would pass, then with a restart of unity, it would fail: It's all very broken. I would do some tests with it, have someone else pull and run the tests, if it still passes then release and just be prepared to change it if it later causes issues.